megamek/src/megamek/client/ui/swing/boardview/EntitySprite.java

```diff
@@ -71,7 +74,14 @@
     
     // Keep track of ECM state, as it's too expensive to compute on the fly.
     private boolean isAffectedByECM = false;
-
+    
+    private static final Set<String> removableNameStrings = new HashSet<String>( 
+            Arrays.asList("Defense", "Platoon", "Heavy", "Medium", "Light", "Artillery", "Tank", 
+                    "Wheeled", "Command", "Standard", "Hover", "Hovercraft", "Mechanized", 
+                    "(Standard)", "Platoon", "Defense", "Transport", "Vehicle", "Air", 
+                    "Assault", "Mobile", "Platform", "Battle Armor", "Vessel", "Infantry",
+                    "Fighting", "Fire", "Suport", "Reconnaissance", "Fast"));
```

Comment:
Why not just keep it as an array? Also, new HashSet<String> is completely superfluous in modern Java, as you only need to specify the class (i.e. String) if it is a wildcard on the backend

Reply:
I'll change the initialization to Set.of(), much simpler. The reduceVehicleName() must find if a term is contained in that word list, which is simplest and fastest in a HashSet (although that millisecond won't be important)